### PR TITLE
Fixing unnecessary request to update string service

### LINKED ISSUE #286: Sending request to change item string when nothing as been changed
When only the item has been changed, and not the item string, not request should be sent to change the item string

### PR DESCRIPTION
**Pull request by AyeRlS**
Fixes #286

### CODE REVIEW

File: src/app/modules/item/pages/item-edit/item-edit.component.ts [hunk 1]
```diff
@@ -127,6 +127,7 @@ export class ItemEditComponent implements OnDestroy, PendingChangesComponent {
 
   // Item string changes
   private getItemStringChanges(): ItemStringChanges | undefined {
+    if (this.itemForm.pristine || !this.itemForm.touched) return {};
```
Comment (smadbe) [inline src/app/modules/item/pages/item-edit/item-edit.component.ts:130]: why both pristine and not touched?

**Review by smadbe — CHANGES_REQUESTED** [review]
Have you tested it? I still see the request. :-(

File: src/app/modules/item/pages/item-edit/item-edit.component.ts (hunk 1 above)
Comment (AyeRlS) [reply src/app/modules/item/pages/item-edit/item-edit.component.ts:130]: For some reason, on the first loading, the itemForm is marked as dirty. So I had to add the !this.itemForm.touched to handle this case.

Comment (smadbe) [reply src/app/modules/item/pages/item-edit/item-edit.component.ts:130]: If marked dirty, why wouldn't the save/cancel display? They are based on dirty. And if the code goes through "markAsPristine" as is not pristine afterwards, something looks suspicious.

**Review by smadbe — CHANGES_REQUESTED** [review]
And you should the same for the other request. If there is nothing to update in children, it is useless to send a PUT with `{}` as data to be changed..

**Comment by smadbe** [comment]
> Have you tested it? I still see the request. :-(

I cannot reproduce my issue, strange... So you can ignore my remark.

File: src/app/modules/item/pages/item-edit/item-edit.component.ts (hunk 1 above)
Comment (AyeRlS) [reply src/app/modules/item/pages/item-edit/item-edit.component.ts]: So in the end, the way the save/cancel display work was conflicting with what I was trying to do. So I've changed the way to check to see if the fields have been changed

**Comment by smadbe** [comment]
It still does the empty update while there is no update to children. Haven't you tried to fix that?
(test: http://dev.algorea.org/branch/update-string-when-needed/fr/#/items/by-id/972755911692173036;attempId=0;path=4702,4102,1980584647557587953/edit - I add and remove a space in the description and save, both strings (let's say that's ok) and children requests are sent)

**Comment by AyeRlS** [comment]
I didn't understood there was also a problem with the other service, I'm going to fix that too.

File: src/app/modules/item/pages/item-edit/item-edit.component.ts [hunk 2]
```diff
@@ -120,30 +120,40 @@ export class ItemEditComponent implements OnDestroy, PendingChangesComponent {
...
+
+    const title = titleControl.value as string;
+    if (title !== initialValues.title) res.title = title.trim();
+
+    const subtitle = subtitleControl.value as string;
+    if (subtitle !== initialValues.subtitle) res.subtitle = subtitle.trim();
+
+    const description = descriptionControl.value as string;
+    if (description !== initialValues.description) res.description = description.trim();
+
```
Comment (smadbe) [inline src/app/modules/item/pages/item-edit/item-edit.component.ts:148]: Don't you loose the fact that when string is empty for subtitle and description, you were pushing "null" ?

Comment (AyeRlS) [reply src/app/modules/item/pages/item-edit/item-edit.component.ts]: Fixed

Comment (smadbe) [reply src/app/modules/item/pages/item-edit/item-edit.component.ts]: 👍  (It was a check you wrote a few weeks ago after many iterations. It should not be me who makes you notice that you do not handle this anymore.)

**Review by smadbe — CHANGES_REQUESTED** [review]
I open a item with an empty description, I add a space, remove it, I still have a request sent... 😔

**Comment by AyeRlS** [comment]
All I'm seeing are OPTIONS and GET requests, what item are you testing this on?

**Comment by smadbe** [comment]
http://dev.algorea.org/branch/update-string-when-needed/fr/#/items/by-id/972755911692173036;attempId=0;path=4702,4102,1980584647557587953/edit 
Screenshot 2021-01-12 at 15 37 56

**Comment by AyeRlS** [comment]
The item I was using for my test was the problem, it should be fixed now.

**Review by smadbe — APPROVED** [review]
Looks ok. I hope you have tested it carefully.

**Pull request merged**